frame/staking/src/lib.rs

```diff
@@ -1363,22 +1363,42 @@ decl_module! {
 			consumed_weight
 		}
 
-		/// Check if the current block number is the one at which the election window has been set
-		/// to open. If so, it runs the offchain worker code.
+		/// Offchain worker logic.
 		fn offchain_worker(now: T::BlockNumber) {
-			use offchain_election::{set_check_offchain_execution_status, compute_offchain_election};
+			use offchain_election::{
+				set_check_offchain_execution_status, compute_save_and_submit,
+				restore_or_compute_then_submit, OFFCHAIN_REPEAT,
+			};
+			// ensure that we don't run OCW in any case more at least with 5 blocks delay.
+			let threshold: T::BlockNumber = OFFCHAIN_REPEAT.into();
 
-			if Self::era_election_status().is_open_at(now) {
-				let offchain_status = set_check_offchain_execution_status::<T>(now);
-				if let Err(why) = offchain_status {
-					log!(warn, "💸 skipping offchain worker in open election window due to [{}]", why);
-				} else {
-					if let Err(e) = compute_offchain_election::<T>() {
-						log!(error, "💸 Error in election offchain worker: {:?}", e);
-					} else {
-						log!(debug, "💸 Executed offchain worker thread without errors.");
+			let election_status = Self::era_election_status();
+
+			log!(
+				trace,
+				"Running OCW at {:?}, election status = {:?}",
+				now,
+				election_status,
+			);
+			match Self::era_election_status() {
+				ElectionStatus::Open(opened) if opened == now => {
+					// If era election status is open at the current block, mine a new solution
+					// then save and submit it.
+					let initial_output = set_check_offchain_execution_status::<T>(now, threshold)
+						.and_then(|_| compute_save_and_submit::<T>());
+					log!(info, "initial OCW output at {:?} = {:?}", now, initial_output);
+				},
+				ElectionStatus::Open(opened) if now > opened => {
```

Comment:
Don't we want to wait for some blocks at least to make sure there was enough time to produce + include a solution?

Reply:
there's a lock that you need to acquire in order to re-run anything, and that should enforce a 5 block delay. I think that is more than enough.